htdocs/class/xoopsform/renderer/XoopsFormRenderer.php

```diff
@@ -40,7 +40,7 @@ final class XoopsFormRenderer
     public static function getInstance()
     {
         if (null === self::$instance) {
-            static::$instance = new static();
+            static::$instance = new self();
```

Comment:
No!! new static. This is counting on late static binding to do the right thing. No "self" here please.